### PR TITLE
Feature/db connection

### PR DESCRIPTION
**Pull request by Djangofs**

### CODE REVIEW

File: src/connectors/registrationDb/registrationDb.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  try {
+    const response = await model.findOne(query);
+    logEmitter.emit(
+      "functionSuccess",
+      "registration.connector.js",
+      functionName
+    );
+    return response;
+  } catch (err) {
+    logEmitter.emit(
```
Comment (Joe-Brady) [inline src/connectors/registrationDb/registrationDb.js]: Should this have an error code etc before throwing?

File: src/connectors/registrationDb/registrationDb.test.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        });
+      });
+    });
+  });
+
+  // get all registrations by council
+  describe("Function: getAllRegistrationsByCouncil", () => {
+    let result;
+    // Should respond with double response in double mode
+    describe("it should response with double response in double mode", () => {
```
Comment (Joe-Brady) [inline src/connectors/registrationDb/registrationDb.test.js]: just 'in double mode' as the describe wording? (Minor point)

File: src/connectors/registrationDb/registrationDb.test.js [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+        expect(Registration.findAll.mock.calls[0][0].where).toEqual({
+          council: "west-dorset"
+        });
+      });
+
+      it("it should call the models the appropriate numbers of times", () => {
+        expect(Operator.findOne.mock.calls.length).toBe(4);
+      });
+    });
+    // Should call find all registrations
```
Comment (Joe-Brady) [inline src/connectors/registrationDb/registrationDb.test.js]: Are these to-dos?

File: src/db/models/index.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+fs.readdirSync(__dirname)
+  .filter(file => {
+    return (
+      file.indexOf(".") !== 0 && file !== basename && file.slice(-3) === ".js"
+    );
+  })
+  .forEach(file => {
+    const model = sequelize["import"](path.join(__dirname, file));
+    db[model.name] = model;
```
Comment (Joe-Brady) [inline src/db/models/index.js]: how have we done this before? Looks like a complex approach

File: src/connectors/registrationDb/registrationDb.js (hunk 1 above)
Comment (Djangofs) [reply src/connectors/registrationDb/registrationDb.js]: Yes! done

File: src/connectors/registrationDb/registrationDb.test.js (hunk 2 above)
Comment (Djangofs) [reply src/connectors/registrationDb/registrationDb.test.js]: yep

File: src/connectors/registrationDb/registrationDb.test.js [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+        expect(Activities.findOne).not.toHaveBeenCalled();
+        expect(Establishment.findOne).not.toHaveBeenCalled();
+        expect(Metadata.findOne).not.toHaveBeenCalled();
+        expect(Operator.findOne).not.toHaveBeenCalled();
+        expect(Premise.findOne).not.toHaveBeenCalled();
+        expect(Registration.findAll).not.toHaveBeenCalled();
+      });
+    });
+
+    describe("it should return all responses when not in double mode", () => {
```
[the remark this answers is not in the document]
Comment (Djangofs) [reply src/connectors/registrationDb/registrationDb.test.js]: yep

File: src/connectors/registrationDb/registrationDb.test.js (hunk 3 above)
Comment (Djangofs) [reply src/connectors/registrationDb/registrationDb.test.js]: yep

**Pull request merged**